src/NUnit.Xml.TestLogger/NUnitXmlTestLogger.cs

```diff
@@ -0,0 +1,501 @@
+﻿namespace Microsoft.VisualStudio.TestPlatform.Extension.NUnit.Xml.TestLogger
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.IO;
+    using System.Linq;
+    using System.Xml.Linq;
+    using System.Text;
+    using System.Text.RegularExpressions;
+    using ObjectModel;
+    using ObjectModel.Client;
+    using ObjectModel.Logging;
+
+    [FriendlyName(FriendlyName)]
+    [ExtensionUri(ExtensionUri)]
+    public class NUnitXmlTestLogger : ITestLoggerWithParameters
+    {
+        /// <summary>
+        /// Uri used to uniquely identify the logger.
+        /// </summary>
+        public const string ExtensionUri = "logger://Microsoft/TestPlatform/NUnitXmlLogger/v1";
+
+        /// <summary>
+        /// Alternate user friendly string to uniquely identify the console logger.
+        /// </summary>
+        public const string FriendlyName = "nunit";
+
+        public const string LogFilePathKey = "LogFilePath";
+        public const string EnvironmentKey = "Environment";
+        public const string NUnitVersionKey = "NUnitVersion";
+
+        private const string ResultStatusPassed = "Passed";
+        private const string ResultStatusFailed = "Failed";
+
+        private string outputFilePath;
+        private string environmentOpt;
```

Comment:
`environmentOpt` and `nunitVersionOpt` is not getting used, Should we remove this?

Reply:
Yes, removed them.